lib/assets/javascripts/src/local-time/element_observations.coffee

```diff
@@ -0,0 +1,52 @@
+import LocalTime from "./local_time"
+
+{elementMatchesSelector} = LocalTime
+
+class LocalTime.ElementObservations
+  OBSERVABLE_ATTRIBUTES = [ "datetime", "data-local", "data-format", "data-format24" ]
+
+  constructor: (@selector, @callback) ->
+    @elements = new Map()
+
+  include: (element) =>
+    unless @elements.get(element)
+      observer = @observe(element)
+      @register(element, observer)
+
+  disregard: (element) =>
+    if observer = @elements.get(element)?.observer
+      observer.disconnect()
+      @elements.delete(element)
+
+  observe: (element) =>
+    observer = new MutationObserver(@processMutations)
+    observer.observe(element, characterData: true, subtree: true, attributes: true, attributeFilter: OBSERVABLE_ATTRIBUTES)
+    observer
+
+  register: (element, observer) =>
+    @elements.set(element, { observer: observer, updates: -1 })
+    @incrementUpdates(element)
+
+  processMutations: (mutations) =>
+    for mutation in mutations
+      target = mutation.target
+
+      element = if target.nodeType is Node.TEXT_NODE
+        target.parentNode
+      else
+        target
```

Comment:
Turbo morphs could change either/or — the data attributes on the element itself, or the value of the nested text node